src/Upscale.cpp

```diff
@@ -244,6 +245,20 @@ void Upscale::Orthogonalize(mfem::BlockVector& vect) const
     Orthogonalize(vect.GetBlock(1));
 }
 
+void Upscale::OrthogonalizeCoarse(mfem::Vector& vect) const
+{
+    const mfem::Vector coarse_constant_rep = GetCoarseConstantRep();
+    double local_dot = (vect * coarse_constant_rep);
+    double global_dot;
+    MPI_Allreduce(&local_dot, &global_dot, 1, MPI_DOUBLE, MPI_SUM, comm_);
+    vect.Add(-global_dot, coarse_constant_rep);
```

Comment:
@chakshinglee Referring to your note from an earlier commit, should this include the fine level global size somewhere to compute average?  Or is that automatically handled already?

`vect.Add(-global_dot / fine_global_size, coarse_constant_rep);`

Reply:
Good catch, you are right. I missed the inner product of constant vector. I actually missed that in the fine level formula, and consequently the coarse one as well (because it was derived from fine). It should be v -= (v * 1 ) / (1 * 1) * 1. Similarly for coarse level v_c -= (v_c * 1_c ) / (1_c * 1_c) * 1_c.

The inner product (1_c * 1_c) should have the same value as (1 * 1), which equals `fine_global_size`. Either include that factor in the normalization, or if we store this constant representation, we can just store the normalized version, i.e. store 1_c / || 1_c ||, then (1_c * 1_c) = 1 and we dont need that factor.